src/dependency-manager/src/manager.ts

```diff
@@ -117,8 +120,8 @@ export default abstract class DependencyManager {
             const param_target_datastore_name = (param_value as DatastoreValueFromParameter).valueFrom.datastore;
 
             if (param_target_service_name) {
-              if (value_from_param.valueFrom.interface && !all_interface_names.includes(value_from_param.valueFrom.interface)) {
-                throw new Error(`Interface ${value_from_param.valueFrom.interface} is not defined on any service.`);
+              if (value_from_param.valueFrom.interface && !interface_map[param_target_service_name].includes(value_from_param.valueFrom.interface)) {
```

Comment:
why do you need the map? you get the node in question the line below.

`if (value_from_param.valueFrom.interface && value_from_param.valueFrom.interface in param_target_service.interfaces)`